pkg/machineproviders/providers/openshift/machine/v1beta1/failuredomain/failuredomain.go

```diff
@@ -47,13 +58,31 @@ type FailureDomain interface {
 
 	// AWS returns the AWSFailureDomain if the platform type is AWS.
 	AWS() machinev1.AWSFailureDomain
+
+	// AWS returns the AzureFailureDomain if the platform type is Azure.
+	Azure() machinev1.AzureFailureDomain
+
+	// GCP returns the GCPFailureDomain if the platform type is GCP.
+	GCP() machinev1.GCPFailureDomain
+
+	// OpenStack returns the OpenStackFailureDomain if the platform type is OpenStack.
+	OpenStack() machinev1.OpenStackFailureDomain
+
+	// Equal compares the underlying failure domain.
+	Equal(other FailureDomain) bool
 }
 
 // failureDomain holds an implementation of the FailureDomain interface.
 type failureDomain struct {
 	platformType configv1.PlatformType
 
 	aws machinev1.AWSFailureDomain
+
+	azure machinev1.AzureFailureDomain
+
+	gcp machinev1.GCPFailureDomain
+
+	openStack machinev1.OpenStackFailureDomain
```

Comment:
If you want you can squash these together a bit
```suggestion
	azure machinev1.AzureFailureDomain
	gcp machinev1.GCPFailureDomain
	openStack machinev1.OpenStackFailureDomain
```